debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/KafkaConnectRecordParser.java

```diff
@@ -189,31 +191,45 @@ protected void parseValueFields(Struct value, Record r) {
         Struct after = value.getStruct("after");
         // TODO: error handle before is NULL
         Struct before = value.getStruct("before");
+        
+        // in case of delete, the before struct contains the values required in cases like DELETE-INSERT conflicts(unique key columns)
+        if (r.op.equals("d") && before != null) {
+            after = before;
+        }
         if (after == null) {
             return;
         }
+
         for (Field f : after.schema().fields()) {
             Object fieldValue;
             if (sourceType.equals("yb")){
                 // TODO: write a proper transformer for this logic
-                // values in the debezium connector are as follows:
-                // "val1" : {
-                //  "value" : "value for val1 column",
-                //  "set" : true
-                //}
-                Struct valueAndSet = after.getStruct(f.name());
-                if (r.op.equals("u")) {
-                    // in the default configuration of the stream, for an update, the fields in the after struct
-                    // are only the delta fields, therefore, it is possible for a field to not be there.
-                    if (valueAndSet == null){
-                        continue;
+                /* 
+                    values in the debezium connector are as follows:
+                    "val1" : {
+                        "value" : "value for val1 column",
+                        "set" : true
                     }
+                */
+                Struct beforeValueAndStruct = null;
+                if (before != null) { // null for INSERTs
+                    beforeValueAndStruct = before.getStruct(f.name());
+                } 
+                Struct afterValueAndSet = after.getStruct(f.name());
+                // there is no valueAndSet for the columns with null values
```

Comment:
Pls document that this is specifically for the INSERT type events that ValueAndSet is null for null values. For all other event types, ValueAndSet is non-null, and value specifically is null. (correct?)

Reply:
In fact, just add a check that `afterValueAndSet` can only be null in case of insert type events.